File: 334.py
class Solution:
    def increasingTriplet(self, nums: list[int]) -> bool:
        for i in range(0, len(nums)-2):
            for j in range(i+1, len(nums)-1):
                for k in range(j+1, len(nums)):
                    if nums[i] < nums[j] and nums[j] < nums[k]:
                        return True
        return False

print(Solution().increasingTriplet([1,2,3,4,5]))

File: 334.go
package main

import (
	"fmt"
	"math"
)

func increasingTriplet(nums []int) bool {
	small := math.MaxInt64
	medium := small

	for i := range nums {
		if nums[i] <= small {
			small = nums[i]
		} else if nums[i] <= medium {
			medium = nums[i]
		} else {
			return true
		}
	}
	return false
}

func main() {
	fmt.Println(increasingTriplet([]int{1, 5, 0, 4, 1, 3}))
}
